### DISCUSSION TITLE
Regional outpaint concept design.

### DISCUSSION THREAD

**Discussion by Symbiomatrix** [hako-mikan/sd-webui-regional-prompter#122] [Ideas]
So this is something that I've attempted back in the early days of webui; I think that applying the method to regions should be possible (hopefully not being too optimistic).
Basic idea is this, in ratio mode: Add negative values along the edges to extend the image in that direction.
- The difference compared to normal ratios, is that the extending quantity is relative to the sum of the positive values. For example, -1, 2, 2, -2 will extend the image by 25% to the left and 50% to the right.
- Multiple negative values can appear sequentially (eg -2, -1, 1), and each will correspond to a different region as usual.
- Rows follow the same rules, extending relative to the sum of positive row ratios.
- ADDROW / COL are unaffected, though might be useful to create OUTROW / COL keys in the same vein.
- I like the idea of being able to modify the image along with the outer edges to an extent (or not at all). Might try to add outer specific steps or str.

Not really sure how best to integrate with mask mode. Second set of colours, maybe, measure relative to central part (bounding box); but what to do with leftovers inside the center, should they be allowed to remain, maintaining the prompt? Is there a simpler way? Use canvas dimensions?

**Comment by hako-mikan** [comment]
You mean adding the regions to exiting image generated by this script? 
![outcol](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/122196982/287fc529-cdc6-48dd-b948-8c74371ea640)

**Reply by Symbiomatrix** [reply]
Aye, pretty much, with a finer grained control over the outer regions, like so (the values are quite inaccurate visually, but imagine the extra column is 50% of the original image):

![RegionalOutpaintB](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/41131377/2c3cad77-6624-454d-8704-e95694069b82)

Essentially, the goal is to be able to generate a whole new image based on the existing one, rather than piecemeal additions like openoutpaint was built to do. That's why being able to vary the existing parts is important, because the composition might change slightly with the outer regions. And in my experience, you must have a very high strength and steps just to get anything meaningful in outer regions, so it makes sense that they should have their own small set of parameters.
